packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -107,11 +107,28 @@ class Directionality extends InheritedWidget {
   /// ```dart
   /// TextDirection textDirection = Directionality.of(context);
   /// ```
+  // TODO(goderbauer): Make this non-null when customers have upgraded to Directionality.maybeOf.
```

Comment:
Should we start encouraging people to migrate by adding an assert here with a message saying to use `maybeOf` if the resolved direction is null?

Reply:
That could be kinda spammy. I think we usually don't do that.

Reply:
Won't that effectively be what users gets though when we actually make this non-null? But probably without the benefit of a message printing next to the error that they really should use the other variant...

Reply:
We can make the error message link to TextDirection.maybeOf.

Reply:
How does that work for compile time errors?

Reply:
What do you mean? The assert added will be a runtime error.

Reply:
We will basically just call https://master-api.flutter.dev/flutter/widgets/debugCheckHasDirectionality.html at the beginning of this method.

Reply:
Discussed offline - most apps will get directionality from material/cupertinoapp, probably not worth worrying about outside of tests.